### PR TITLE
Fix #89791 - Reordering parts and undo actions

### PR DESCRIPTION
**Pull request by vgstef**
An important rewrite of the excerpts dialog actions following discussions on https://musescore.org/fr/node/89791

- This PR makes all actions (delete, add, reorder) to be effective only when we click the Ok button.
- This change makes it possible to fix the reordering issues. 
- All actions (delete, add, reorder) done by the user in the dialog can now be undone (or redone) in a single undo command.
- As a side effect, it also fixes the dialog window flickering that was happening when deleting a part.

I've tested a lot of use cases, and everything works.

I've put many comments the code and I kept the different "pass" separated. So this could be made more compact if it too lengthy for the MS coding rules...

### CODE REVIEW

**Comment by ZackTheCardshark** [comment]
It appears there is a critical bug to be ironed out here. To reproduce, open any existing score with linked parts (or create and save one with "New All"), then add one new part containing more than one instrument, checking the boxes in bottom-to-top order. Attempt to save, or wait for the autosave. MuseScore crashes, and the autorecovery file is corrupted.

**Comment by vgstef** [comment]
Thanks Zack for trying it out. I'm sorry, but I can't reproduce. I tried many different combinations, and it never crashes.

Have you tried to your issue with the 2.0.2 version to see if your bug was already there? Because I don't see why my PR would have introduced such a crash.

Also, could your give a step by step procedure to reproduce, and maybe an example file?

**Comment by ZackTheCardshark** [comment]
Ah yes, same problem in 2.0.2. See https://musescore.org/en/node/91271.

File: libmscore/excerpt.cpp [hunk 1]
```diff
@@ -86,6 +86,21 @@ bool Excerpt::operator!=(const Excerpt& e) const
       }
 
 //---------------------------------------------------------
+//   operator!==
```
Comment (lasconic) [inline libmscore/excerpt.cpp]: Comment is wrong here.

**Comment by lasconic** [comment]
Could you squash the commits together? http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Pull request merged**